### PR TITLE
Finally update the API V2 instructions

### PR DESCRIPTION
**Pull request by plaindocs**
Fixes #909 and #880 

FAO @iriberri 

Relevant to
- #1047 
- #303 

I'm gonna merge this in quicksharp I think, so I can make a start on the next steps. EDIT Or just work on this branch...

### CODE REVIEW

**Comment by plaindocs** [comment]
Up to about here is all fixing V2 stuff. Looking at V2.1 changes from here on.

File: user/developer.md [hunk 1]
```diff
@@ -27,6 +27,22 @@ We've created an [API Explorer](https://developer.travis-ci.org/) that closely
...
 
+## API V2.1
+
+We've released an update to the Travis CI API V2, which is API V2.1. This update allows users of Travis CI for private projects to make certain build information available through the API.
+
+For users of Travis CI for public repositories there is no change.
+
+API V2.1 is identical to API V2 **except for the following breaking changes**:
+
+* For public repositories, unauthenticated requests receive an HTTP 200 or 400 error.
```
Comment (aakritigupta) [inline user/developer.md]: According to Sven's docs here, the return status codes can be 200 or 404 https://gist.github.com/svenfuchs/5df1b12efd607d8914ceb7cd3dadad93

File: user/developer.md [hunk 2]
```diff
@@ -27,6 +27,22 @@ We've created an [API Explorer](https://developer.travis-ci.org/) that closely
...
+* For private repositories, unauthenticated requests receive an HTTP 200 error.
+* For private repositories, authenticated requests by users that do not have permission to view the repository receive an HTTP 400 error.
```
Comment (aakritigupta) [inline user/developer.md]: Should be 404 error, not 400

File: user/developer.md (hunk 1 above)
Comment (aakritigupta) [reply user/developer.md]: Fixed

File: user/developer.md [hunk 3]
```diff
@@ -27,6 +27,22 @@ We've created an [API Explorer](https://developer.travis-ci.org/) that closely
...
+## API V2.1
+
+We've released an update to the Travis CI API V2, which is API V2.1. This update allows users of Travis CI for private projects to make certain build information available through the API.
+
+For users of Travis CI for public repositories there is no change.
+
+API V2.1 is identical to API V2 **except for the following breaking changes**:
+
+* For public repositories, unauthenticated requests receive an HTTP 200 or 400 error.
+* For private repositories, unauthenticated requests receive an HTTP 200 error.
```
[the remark this answers is not in the document]
Comment (aakritigupta) [reply user/developer.md]: Fixed

File: user/developer.md (hunk 2 above)
Comment (aakritigupta) [reply user/developer.md]: Fixed

File: user/developer.md [hunk 4]
```diff
@@ -27,6 +27,24 @@ We've created an [API Explorer](https://developer.travis-ci.org/) that closely
 integrates with V3, updating automatically when new endpoints are added, and
 includes a useful tool for exploring endpoints.
 
+## API V2.1
+
+We've released an update to the Travis CI API V2, which is API V2.1. This update allows users of Travis CI for private projects to make certain build information available through the API.
+
+For users of Travis CI for public repositories there is no change.
```
Comment (iriberri) [inline user/developer.md]: Should we say something like: "There is no change for our API for Open Source projects built at api.travis-ci.org" or similar?

Comment (aakritigupta) [reply user/developer.md]: Fixed

**Comment by aakritigupta** [comment]
Leaving this PR to be merged by @iriberri when the blog post for announcing v2.1 is ready.

I suggest the docs be released 1-2h before the blog post to make sure everything works just fine with slate.

**Comment by svenfuchs** [comment]
I think this is much better now, @aakritigupta 

Couple more thoughts/questions:

We don't really have any overview content, do we? We talk about the versions under "Overview". Should we rename that headline to "Versions"?

I think the paragraph:

> As a stepping stone to new and greater things, we’ve released an update to the Travis CI API V2, which is API V2.1.

should probably sit under the headline "API V2.1", not "API V3"?

I'd personally remove the note "Note that the rest of the examples in this document have the Accept header set to API V2." ... and instead replace the version header with the one we now recommend, i.e. replace `application/vnd.travis-ci.2+json` with `application/vnd.travis-ci.2.1+json` throughout the entire documentation?

**Comment by aakritigupta** [comment]
@svenfuchs fixed the v2.1 announcement positioning.

I think we can push this PR out once the blog post is out, and the v3 examples can be done in a separate PR. But I will leave it to @plaindocs to decide. He is back tomorrow :)

**Comment by aakritigupta** [comment]
The build is only failing because www.graphviz.org is returning a 404 and we use this link in one of our examples.
All remaining tests run locally.

**Comment by aakritigupta** [comment]
> I'd personally remove the note "Note that the rest of the examples in this document have the Accept header set to API V2." ... and instead replace the version header with the one we now recommend, i.e. replace application/vnd.travis-ci.2+json with application/vnd.travis-ci.2.1+json throughout the entire documentation?

On second thoughts about this - since we have an API explorer for v3, it may not make much sense to write up code based documentation in slate again for v3. I'd be fine with leaving the code examples in v2 on this page. It already links to V3 right at the top.

**Comment by iriberri** [comment]
Thanks for the fixes, @aakritigupta! :)

As Sven mentions, I'm on the side of updating all the V2 example headers to go to `application/vnd.travis-ci.2.1+json` instead of V2 too. I'll be happy to help with this if needed!

**Comment by plaindocs** [comment]
Hey folks, just checking in back from vacation. I'll catch up and be in touch today / tomorrow.

**Comment by plaindocs** [comment]
Regarding

> On second thoughts about this - since we have an API explorer for v3, it may not make much sense to write up code based documentation in slate again for v3. I'd be fine with leaving the code examples in v2 on this page. It already links to V3 right at the top.

I'd prefer to leave the Slate docs referring to 2 / 2.1 and use the new explorer for 3.

**Comment by aakritigupta** [comment]
@plaindocs We should still update the code examples with api v2.1 instead of v2 though
 I agree with:
>As Sven mentions, I'm on the side of updating all the V2 example headers to go to application/vnd.travis-ci.2.1+json instead of V2 too. I'll be happy to help with this if needed!

**Comment by plaindocs** [comment]
@aakritigupta this is fine to merge from my side.

**Comment by iriberri** [comment]
As suggested by Sam, I brought all these commits to https://github.com/travis-ci/docs-travis-ci-com/pull/1736, so I guess we can close this PR and merge this through the other.

**Pull request merged**